kraken-build/src/kraken/std/python/tasks/publish_task.py

```diff
@@ -31,47 +37,157 @@ def get_relationships(self) -> Iterable[TaskRelationship]:
+        existing_files = _get_existing_files_from_index(
+            project_name, self.index_index_url.get(), self.index_credentials.get()
+        )
+        logger.debug(f"Existing files: {existing_files}")
+
+        # If we can't check, proceed to execute and let uv handle it.
+        if existing_files is None:
+            return None
+
+        files_to_publish = [dist for dist in distributions if dist.name not in existing_files]
+
+        if not files_to_publish:
+            return TaskStatus.skipped("All distribution files already exist on the index.")
+
+        # Update the distributions to only publish the ones that don't exist.
+        self.distributions.set(files_to_publish)
+        return None
+
     def execute(self) -> TaskStatus:
+        # Check for the deprecated property
+        if self.interactive.get() is not None:
+            self.logger.warning(
+                "The 'interactive' property on the python.publish task is deprecated and has no effect. "
+                "uv publish is non-interactive by default in this context."
+            )
         credentials = self.index_credentials.get()
-        repository_url = self.index_upload_url.get().rstrip("/") + "/"
         command = [
-            *self.twine_cmd.get(),
-            "upload",
-            "--repository-url",
-            repository_url,
-            "--verbose",
-            *[str(x.absolute()) for x in self.distributions.get()],
+            "uv",
+            "publish",
+            "--no-progress",  # No spinners and progress bars in stderr
+            "--publish-url",
+            self.index_upload_url.get(),
         ]
+        distributions = self.distributions.get()
+        if not distributions:
+            return TaskStatus.succeeded("No new distributions to publish.")
+
+        command.extend([str(x.absolute()) for x in distributions])
+
+        env = {}
         if credentials:
-            command += [
-                "--username",
-                credentials[0],
-                "--password",
-                credentials[1],
-            ]
-        if not self.interactive.get():
-            command.append("--non-interactive")
-        if self.skip_existing.get():
-            command.append("--skip-existing")
-
-        safe_command = [x.replace(credentials[1], "MASKED") for x in command] if credentials else command
+            if "pypi.org/" in self.index_upload_url.get() and credentials[0] != "__token__":
+                self.logger.warning(
+                    "Since 2024-01-01, PyPI no longer allows publishing with username and password, "
+                    " see https://blog.pypi.org/posts/2024-01-01-2fa-enforced/"
+                    " and https://docs.astral.sh/uv/guides/package/#publishing-your-package"
+                )
+            env["UV_PUBLISH_USERNAME"] = credentials[0]
+            env["UV_PUBLISH_PASSWORD"] = credentials[1]
+
+        safe_command = command
         self.logger.info("$ %s", safe_command)
```

Comment:
```suggestion
        safe_command = command
        self.logger.info("$ %s", command)
```

Since the command itself does not contain unsafe data